src/open_inwoner/pdc/managers.py

```diff
@@ -33,6 +37,49 @@ def visible_for_user(self, user: User):
             return self.filter(visible_for_authenticated=True)
         return self.filter(visible_for_anonymous=True)
 
+    def filter_for_user_with_zaken(self, user: User):
+        """
+        Returns the categories linked to ZaakTypen for which the user has Zaken.
+        """
+        if not getattr(user, "bsn", None):
+            return self
+
+        cases = fetch_cases(user.bsn)
+
+        months_since_last_zaak_per_zaaktype = {}
+        for case in cases:
+            resolve_zaak_type(case)
+
+            duration_since_start = relativedelta(date.today(), case.startdatum)
+            if (
+                case.zaaktype.identificatie not in months_since_last_zaak_per_zaaktype
+                or months_since_last_zaak_per_zaaktype[case.zaaktype.identificatie]
+                > duration_since_start.months
+            ):
+                months_since_last_zaak_per_zaaktype[
+                    case.zaaktype.identificatie
+                ] = duration_since_start.months
+
+        zaaktype_ids = list(months_since_last_zaak_per_zaaktype.keys())
+
+        qs = self.filter(zaaktypen__overlap=zaaktype_ids)
```

Comment:
@alextreme I assume that only categories linked to zaaktypen will be shown in the highlighted category plugin for DigiD authenticated users

Reply:
That is correct, however note that not all gemeentes will use this feature (for those the categories shown should be as-is). We could make this a featureflag if necessary in the algemene configuratie.

Reply:
Good point, I'll add a feature flag for this, otherwise the other visibility flags are overridden by this one